torch/_inductor/scheduler.py

```diff
@@ -166,6 +200,14 @@ class BaseSchedulerNode:
     # .min_order = .max_order = X if this node is X-th node in `self.scheduler.nodes`.
     min_order: int
     max_order: int
+    pred_buffers: List[Union[SchedulerBuffer, InputBuffer]]
+    pred_nodes: List[BaseSchedulerNode]
+    succ_nodes: List[BaseSchedulerNode]
+    indegree: int
+    index: int
+    memory_to_free: int
+    size: int
+    measure: int
```

Comment:
Do these new new fields have to live in `BaseSchedulerNode` object (and in `SchedulerBuffer` object), or can we use dictionaries to represent them? Reason is that they are only used in the reorder for peak memory pass and would be great to make them self-contained within the pass.

Reply:
Adding them here to deal with mypy linter complains. But it seems that once I moved the changes to a separate file memory.py, this complaint is gone. So will remove them from here.

Reply:
nvm, the complaint is still there and it just took mypy a while to pick them up.

Reply:
I set up two classes to record these information, one called `MemoryPlanningInfoForBuffer` and one called `MemoryPlanningInfoForNode`. Putting them in `scheduler.py` right now instead of `memory.py` to avoid circular imports.